lib/base/process.cpp

```diff
@@ -101,8 +101,16 @@ static Value ProcessSpawnImpl(struct msghdr *msgh, const Dictionary::Ptr& reques
 
 	auto **envp = new char *[envc + (extraEnvironment ? extraEnvironment->GetLength() : 0) + 2];
 
-	for (int i = 0; i < envc; i++)
+	for (int i = 0; i < envc; i++) {
+		String str(environ[i]);
```

Comment:
`String` allocates memory which you don't need here.

Find a pure C function which does what you need.